addons/l10n_jo_edi/models/account_edi_xml_ubl_21_jo.py

```diff
@@ -0,0 +1,360 @@
+                    'currency_dp': self._get_currency_decimal_places(),
+                    'taxable_amount': vals['raw_base_amount'],
+                    'tax_amount': vals['raw_tax_amount'],
+                    'tax_category_vals': vals['_tax_category_vals_'],
+                }
+                tax_totals_vals['tax_subtotal_vals'].append(subtotal)
+                tax_totals_vals['tax_amount'] += subtotal['tax_amount']
+
+        return [tax_totals_vals]
+
+    def _get_invoice_tax_totals_vals_list(self, invoice, taxes_vals):
+        if invoice._get_jo_invoice_type_number() in [1, 2]:
+            return []
+
+        vals = self._get_invoice_tax_totals_vals_list_helper(invoice, taxes_vals)
+        if invoice._get_jo_invoice_type_number() != 4:
+            vals[0]['tax_subtotal_vals'] = []
+        return vals
+
+    def _get_invoice_line_item_vals(self, line, taxes_vals):
+        product = line.product_id
+        description = line.name and line.name.replace('\n', ', ')
+        return {
+            'name': product.name or description,
+        }
+
+    def _get_document_allowance_charge_vals_list(self, invoice, taxes_vals):
+        discount_amount = 0
+        invoice_lines = invoice.invoice_line_ids.filtered(lambda line: line.display_type not in ('line_note', 'line_section'))
+        for line in invoice_lines:
+            line_taxes_vals = taxes_vals['tax_details_per_record'][line]
+            discount_amount += self._get_line_discount_jod(line, line_taxes_vals)
+        return [{
+            'charge_indicator': 'false',
+            'allowance_charge_reason': 'discount',
+            'currency_name': JO_CURRENCY.name,
+            'currency_dp': self._get_currency_decimal_places(),
+            'amount': discount_amount,
+        }]
+
+    def _get_line_amount_before_discount_jod(self, line, taxes_vals):
+        amount_after_discount = taxes_vals['base_line']['tax_details']['raw_total_excluded']
+        return amount_after_discount / (1 - line.discount / 100)
+
+    def _get_line_discount_jod(self, line, taxes_vals):
+        return self._get_line_amount_before_discount_jod(line, taxes_vals) * line.discount / 100
+
+    def _get_unit_price_jod(self, line, taxes_vals):
+        return self._get_line_amount_before_discount_jod(line, taxes_vals) / line.quantity
+
+    def _get_invoice_line_allowance_vals_list(self, line, taxes_vals):
+        return [{
+            'charge_indicator': 'false',
+            'allowance_charge_reason': 'DISCOUNT',
+            'currency_name': JO_CURRENCY.name,
+            'currency_dp': self._get_currency_decimal_places(),
+            'amount': self._get_line_discount_jod(line, taxes_vals),
+        }]
+
+    def _get_invoice_line_price_vals(self, line, taxes_vals):
+        return {
+            'currency': JO_CURRENCY,
+            'currency_dp': self._get_currency_decimal_places(),
+            'price_amount': self._get_unit_price_jod(line, taxes_vals),
```

Comment:
I suppose the main difference with the standard UBL is that it is always in JOD

Reply:
Yeah I think so. They really got a standard and customized it very heavily that it's no longer a standard :"D